sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -719,15 +724,17 @@ qualifiedName
     ;
 
 identifier
-    : strictIdentifier
+    : {ansi}? ansiReserved
+    | strictIdentifier
     | ANTI | FULL | INNER | LEFT | SEMI | RIGHT | NATURAL | JOIN | CROSS | ON
     | UNION | INTERSECT | EXCEPT | SETMINUS
```

Comment:
since we have `ansiNonReseved` and `nonReserved`. shall we create a `reserved` and put these keywords there?

Reply:
ok, I'll try to update.

Reply:
Oh, these keywords are not reserved in the master: https://github.com/apache/spark/blob/master/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/TableIdentifierParserSuite.scala#L82

If we move these keywords, not `reserved` but `nonReserved`?

Reply:
are you sure? it looks to me that, these keywords can't be `strictIdentifier`, so we can't put them in `nonReserved`.

Reply:
ah, yes. we can't put these in `nonReserved`. Is it just a naming issue? 
Like this?
```
identifier
    : {ansi}? ansiReserved
    | strictIdentifier
    | {!ansi}? strictNonReserved
    ;

strictNonReserved:
    : ANTI | FULL | INNER | LEFT | SEMI | RIGHT | NATURAL | JOIN | CROSS | ON
    | UNION | INTERSECT | EXCEPT | SETMINUS
    ;
```